exporter/elasticsearchexporter/integrationtest/exporter_bench_test.go

```diff
@@ -5,76 +5,154 @@ package integrationtest
 
 import (
 	"context"
+	"fmt"
 	"sync/atomic"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/exporter/exportertest"
 	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
 )
 
-func BenchmarkLogsExporter(b *testing.B) {
-	for _, tc := range []struct {
-		name      string
-		batchSize int
-	}{
-		{name: "small_batch", batchSize: 10},
-		{name: "medium_batch", batchSize: 100},
-		{name: "large_batch", batchSize: 1000},
-		{name: "xlarge_batch", batchSize: 10000},
-	} {
-		b.Run(tc.name, func(b *testing.B) {
-			benchmarkLogs(b, tc.batchSize)
-		})
+func BenchmarkExporter(b *testing.B) {
+	for _, eventType := range []string{"logs", "traces"} {
+		for _, tc := range []struct {
+			name      string
+			batchSize int
+		}{
+			{name: "small_batch", batchSize: 10},
+			{name: "medium_batch", batchSize: 100},
+			{name: "large_batch", batchSize: 1000},
+			{name: "xlarge_batch", batchSize: 10000},
+		} {
+			b.Run(fmt.Sprintf("%s/%s", eventType, tc.name), func(b *testing.B) {
+				switch eventType {
+				case "logs":
+					benchmarkLogs(b, tc.batchSize)
+				case "traces":
+					benchmarkTraces(b, tc.batchSize)
+				}
+			})
+		}
 	}
 }
 
 func benchmarkLogs(b *testing.B, batchSize int) {
-	var generatedCount, observedCount atomic.Uint64
-
-	receiver := newElasticsearchDataReceiver(b)
-	factory := elasticsearchexporter.NewFactory()
-
-	cfg := factory.CreateDefaultConfig().(*elasticsearchexporter.Config)
-	cfg.Endpoints = []string{receiver.endpoint}
-	cfg.Flush.Interval = 10 * time.Millisecond
-	cfg.NumWorkers = 1
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	exporter, err := factory.CreateLogsExporter(
-		context.Background(),
-		exportertest.NewNopCreateSettings(),
-		cfg,
+	runnerCfg := prepareBenchmark(b, batchSize)
+	exporter, err := runnerCfg.factory.CreateLogsExporter(
```

Comment:
existing bug: this is missing an exporter.Start

Reply:
I didn't call start because we don't pass any `StartFunc` and it was needed so far. The benchmarks would run correctly without start. But now I realize that you are working on batchsender and that would be started in the exporter logic ([ref](https://github.com/open-telemetry/opentelemetry-collector/blob/23b256ca07c3e36cbcd9c19012050251743aea2a/exporter/exporterhelper/common.go#L310-L323)).

Let me add it now.